sdl_ios/SmartDeviceLink/SDLDataStreamHandlingDelegate.h

```diff
@@ -0,0 +1,14 @@
+//
+//  SDLDataStreamHandlingDelegate.h
+//  SmartDeviceLink
+//
+
+#import <Foundation/Foundation.h>
+#import "SDLAbstractProtocol.h"
+
+@interface SDLDataStreamHandlingDelegate : NSObject <NSStreamDelegate>
+
+@property (assign) SDLServiceType serviceType;
+@property (strong, nonatomic) SDLAbstractProtocol *protocol;
```

Comment:
If this is strongly referenced by the proxy, then it could be weak in here, and just test to make sure it doesn't go to nil before using it.